shtdlib.sh

```diff
@@ -419,18 +419,36 @@ function exit_on_fail {
 # add_on_mod callback "${path_to_monitor}"
 #
 function add_on_mod {
-    assert whichs inotifywait
+    if whichs inotifywait ; then
+        file_monitor_command="inotifywait --monitor --recursive --format %w%f
+                                   --event modify
+                                   --event close_write
+                                   --event moved_to
+                                   --event create
+                                   --event moved_from
+                                   --event delete
+                                   --event move_self
+                                   --event delete_self
+                                   --event unmount"
+    elif whichs fswatch ; then
+        file_monitor_command="fswatch  --recursive --format %p
+                                   --event Created
+                                   --event Updated
+                                   --event Removed
+                                   --event Renamed
+                                   --event MovedFrom
+                                   --event MovedTo"
+    else
+        color_echo red "Unable to find inotifywait or fswatch, please install one or the other before trying to use ${0}"
```

Comment:
`${@}` outputs the shell name. 

Suggestion:
```bash
@@ -439,7 +439,7 @@ function add_on_mod {
                                    --event MovedFrom
                                    --event MovedTo"
     else
-        color_echo red "Unable to find inotifywait or fswatch, please install one or the other before trying to use ${0}"
+        color_echo red "Unable to find inotifywait or fswatch, please install one or the other before trying to use '${FUNCNAME[0]} ${*}'"
         return 1
     fi
     local arguments=("${@}")
```

Reply:
$@ == all arguments including the name of the shell script or if used in the context of a function the function name, if in an interactive shell then the name of the shell
$* == all argumentsincluding the name of the shell script or if used in the context of a function the function name, if in an interactive shell then the name of the shell
$0 == First argument, this is the name of the script  or if used in the context of a function the function name, if in an interactive shell then the name of the shell

Reply:
Yup